src/redux/books/books.js

```diff
@@ -25,6 +27,62 @@ const addBookAction = (book) => ({ type: ADD_BOOK, payload: { book } });
 
 const removeBookAction = (id) => ({ type: REMOVE_BOOK, payload: { id } });
 
+const setBooksAction = (books) => ({ type: SET_BOOKS, payload: { books } });
+
+const fetchBooks = () => async (dispatch) => {
+  await fetch(API_LINK)
+    .then((result) => result.json())
+    .then((res) => {
+      const books = Object.entries(res).map((entry) => ({
+        ...entry[1][0],
+        id: entry[0],
+      }));
+      dispatch(setBooksAction(books));
+    });
+};
```

Comment:
- [ ] Good job using fetch to make your API calls 👍🏽. I think it will be better if you use `createAsyncThunk` with a fetch to make your API calls as specified in the project requirement. Using `createAsyncThunk`  makes it easier to create `action creators` for asynchronous API requests which will return a promise for you to manipulate. 

checkout this [Link](https://redux-toolkit.js.org/api/createAsyncThunk) for more details 👍🏽